app/state_machines/form_state_machine.rb

```diff
@@ -0,0 +1,29 @@
+module FormStateMachine
+  extend ActiveSupport::Concern
+
+  included do
+    include AASM
+
+    enum :state, {
+      draft: "draft",
+      live: "live",
+    }
+
+    aasm column: :state, enum: true do
+      state :draft, initial: true
+      state :live
+
+      event :make_form_live do
+        transitions from: :draft, to: :live, guard: Proc.new { task_status_service.mandatory_tasks_completed? }
+      end
+
+      event :make_changes_live do
+        transitions from: :live, to: :draft, guard: Proc.new { has_live_version }
```

Comment:
we do have business rules split across forms-admin and forms-api. For example in forms-admin, if live forms submission email changes when changes are made live then we notify the original submission email address.